msal/src/main/java/com/microsoft/identity/client/NativeAuthPublicClientApplication.kt

```diff
@@ -111,5 +185,381 @@ class NativeAuthPublicClientApplication(
+                    SignInResult.BrowserRequired(
+                        error = BrowserRequiredError(
+                            correlationId = result.correlationId
+                        )
+                    )
+                }
+            }
+        }
+    }
+
+    interface SignInUsingPasswordCallback : Callback<SignInUsingPasswordResult>
+
+    /**
+     * Sign in the account using username and password; callback variant.
+     *
+     * @param username username of the account to sign in.
+     * @param password password of the account to sign in.
+     * @param scopes (Optional) list of scopes to request.
+     * @param callback [com.microsoft.identity.client.NativeAuthPublicClientApplication.SignInUsingPasswordCallback] to receive the result.
+     * @return [com.microsoft.identity.client.statemachine.results.SignInUsingPasswordResult] see detailed possible return state under the object.
+     * @throws MsalClientException if an account is already signed in.
+     */
+    override fun signInUsingPassword(
+        username: String,
+        password: CharArray,
+        scopes: List<String>?,
+        callback: SignInUsingPasswordCallback
+    ) {
+        LogSession.logMethodCall(TAG, "${TAG}.signIn(username: String, password: String, scopes: List<String>?, callback: SignInUsingPasswordCallback)")
+        pcaScope.launch {
+            try {
+                val result = signInUsingPassword(username, password, scopes)
+                callback.onResult(result)
+            } catch (e: MsalException) {
+                Logger.error(TAG, "Exception thrown in signInUsingPassword", e)
+                callback.onError(e)
+            }
+        }
+    }
+
+    /**
+     * Sign in the account using username and password; Kotlin coroutines variant.
+     *
+     * @param username username of the account to sign in.
+     * @param password password of the account to sign in.
+     * @param scopes (Optional) list of scopes to request.
+     * @return [com.microsoft.identity.client.statemachine.results.SignInUsingPasswordResult] see detailed possible return state under the object.
+     * @throws MsalClientException if an account is already signed in.
+     */
+    override suspend fun signInUsingPassword(
+        username: String,
+        password: CharArray,
+        scopes: List<String>?
+    ): SignInUsingPasswordResult {
+        LogSession.logMethodCall(TAG, "${TAG}.signInUsingPassword")
+        return withContext(Dispatchers.IO) {
+            LogSession.logMethodCall(TAG, "${TAG}.signInUsingPassword.withContext")
+
+            val doesAccountExist = checkForPersistedAccount().get()
+            if (doesAccountExist) {
+                Logger.error(
+                    TAG,
+                    "An account is already signed in.",
+                    null
+                )
+                throw MsalClientException(
+                    MsalClientException.INVALID_PARAMETER,
```

Comment:
Is INVALID_PARAMETER the right error in this case?

Reply:
This is the most suitable error from existing codes.